jikesrvm/rvm/src/org/jikesrvm/mm/mminterface/CollectorThread.java

```diff
@@ -42,7 +42,7 @@
   private Address workerInstance = Address.zero();
 
   public void setWorker(Address worker) {
-    rvmThread.assertIsCollector();
+    rvmThread.assertIsRustMMTkCollector();
```

Comment:
I don't understand the intention behind this change. I checked the upstream JikesRVM. Basically we added a field called `isCollector`, and you renamed it to `isRustMMTkCollector`. This means the code in the JikesRVM fork explicitly mentions Rust MMTk. I thought we are going to add support for a general third party heap to JikesRVM. And making it explicitly about Rust MMTk is against our design.

Reply:
Yes, I did this renaming before our latest discussions on the upstreaming.
I can change this to e.g. `assertIsTPHCollector`, but there are other places where the same type of renaming is required before we can upstream (I haven't changed any other RustMMTk related names to TPH).
So, probably for the sake of consistency, we can keep the current name consistent with other name we currently have and change all at once in the future.

The most important issue being addressed in this PR is to move forward to the latest JikesRVM commit so that we need minimal changes to the binding when we upstream.

Reply:
Yes. As I said in the review, the comments don't need to be addressed now.